httpx/config.py

```diff
@@ -352,6 +355,73 @@ def __repr__(self) -> str:
         )
 
 
+class Retries:
+    """
+    Retries configuration.
+
+    Defines the maximum amount of connection failures to retry on, and
+    implements a configurable exponential backoff algorithm.
+    """
+
+    _RETRYABLE_EXCEPTIONS: typing.Sequence[typing.Type[HTTPError]] = (
+        ConnectTimeout,
+        PoolTimeout,
+        NetworkError,
+    )
+    _RETRYABLE_METHODS: typing.Container[str] = frozenset(
+        ("HEAD", "GET", "PUT", "DELETE", "OPTIONS", "TRACE")
+    )
+
+    def __init__(self, retries: RetriesTypes, *, backoff_factor: float = None) -> None:
+        if isinstance(retries, int):
+            limit = retries
+        else:
+            assert isinstance(retries, Retries)
+            assert backoff_factor is None
+            backoff_factor = retries.backoff_factor
+            limit = retries.limit
+
+        if backoff_factor is None:
+            backoff_factor = 0.2
+
+        assert limit >= 0
+        assert backoff_factor > 0
+
+        self.limit: int = limit
+        self.backoff_factor: float = backoff_factor
+
+    def __eq__(self, other: typing.Any) -> bool:
+        return (
+            isinstance(other, Retries)
+            and self.limit == other.limit
+            and self.backoff_factor == other.backoff_factor
+        )
+
+    @classmethod
+    def should_retry_on_exception(cls, exc: HTTPError) -> bool:
+        for exc_cls in cls._RETRYABLE_EXCEPTIONS:
+            if isinstance(exc, exc_cls):
+                break
+        else:
+            return False
```

Comment:
I'm always confused by `for..else` constructs, I always have look it up. In this case I feel this might be clearer:

```python
if any(isinstance(exc, exc_class) for exc_cls in cls._RETRYABLE_EXPECTIONS):
    return False
```

Reply:
Fair enough! Although in this case that'd have be `if not any(...)` :-)